### PR TITLE
Initial deployment subset creation

### PR DESCRIPTION
**Pull request by ronnno**

### CODE REVIEW

File: contracts/Protocol.sol [hunk 1]
```diff
@@ -7,35 +7,42 @@ contract Protocol is IProtocol, Ownable {
...
         }
+        require(protocolVersion > deploymentSubsets[deploymentSubset].currentVersion, "protocol version must be later than current version");
 
         deploymentSubsets[deploymentSubset].nextVersion = protocolVersion;
         deploymentSubsets[deploymentSubset].asOfBlock = asOfBlock;
 
         emit ProtocolVersionChanged(deploymentSubset, protocolVersion, asOfBlock);
     }
+
+    // TODO what about cancelling a future upgrade?
```
Comment (ronnno) [inline contracts/Protocol.sol:47]: this is a good point, maybe we should consider adding a `abortScheduledUpgrade(string calldata deploymentSubset) external onlyOwner`? @OdedWx

Comment (ronnno) [reply contracts/Protocol.sol:47]: discussion summary - @OdedWx agrees we should be able to cancel an upgrade by passing the current protocol version  to `setProtocolVersion` - minor detail to resolve - the resulting event should be emitted with the current block number in `asOfBlock` no matter what block number was really sent in. Do we fail this call when sent with `asOfBlock` in the past?

Comment (vistra) [reply contracts/Protocol.sol:47]: It's not a special case - we pass the current protocol version along with a (near-by) future block, the behaviour remains the same. We only need to allow setting to current version.

File: contracts/Protocol.sol [hunk 2]
```diff
@@ -7,35 +7,42 @@ contract Protocol is IProtocol, Ownable {
...
-                deploymentSubsets[deploymentSubset].currentVersion = deploymentSubsets[deploymentSubset].nextVersion;
-            }
-
-            require(asOfBlock > block.number, "protocol update can only be scheduled for a future block");
-            require(protocolVersion > deploymentSubsets[deploymentSubset].currentVersion, "protocol version must be later than current version");
+        require(deploymentSubsets[deploymentSubset].exists, "deployment subset does not exist");
+        require(asOfBlock > block.number, "protocol update can only be scheduled for a future block");
+
+        if (deploymentSubsets[deploymentSubset].asOfBlock <= block.number) {
+            deploymentSubsets[deploymentSubset].currentVersion = deploymentSubsets[deploymentSubset].nextVersion;
```
[the remark this answers is not in the document]
Comment (vistra) [reply contracts/Protocol.sol:37]: We can, but then we'd have to check that `asOfBlock != 0` when reading it.

**Pull request merged**